### PR TITLE
Add sugar trigraphs for real

### PR DESCRIPTION
**Pull request by Lyxal**

### CODE REVIEW

**Comment by Lyxal** [comment]
At some stage, this needs to automatically print out what the converted program is if it uses sugar trigraphs. That can probably go in the apply section with a boolean set to true if a sugar trigraph is ever processed

**Comment by Lyxal** [comment]
Also as a general naming rule thing:

- `#.<letter>` is the updot version of that letter if it exists
- `#,<letter>` is the downdot version of that letter if it exists
- `#^<letter>` is a superscript version of a letter if it exists
- `#.<symbol>` is what one would expect a trigraph to logically turn into
- `#,<symbol>` is what one would expect a trigraph to aesthetically turn into. Alternatively, if it has an underbar, use that.
`#^<symbol>` is superscript version if it exists. Otherwise, it's a sort of miscellaneous namespace.
`#^.`, `#^:` and `#^%` were chosen for the single, double and triple element grouping modifiers respectively because they have one, two and three circles/lines respectively
L

**Comment by Lyxal** [comment]
Also, trigraphs are subject to change. If any of y'all feel like there's something more suitable for a Unicode, feel free to suggest it.

**Comment by Steffan153** [comment]
We should add tab completion to convert these to their unicode too. Except then maybe the `#` shouldn't be required.

File: shared/src/main/scala/SugarMap.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package vyxal
+object SugarMap:
+  def internalMap: Map[String, String] =
+    val out = Map.newBuilder[String, String]
+
+    val lowerUpdot = "abcdefghlmnoprsx" -> "ȧḃċḋėḟġḣŀṁṅȯṗṙṡṫẋ"
+    val upperUpdot = "ABCDEFGHLMNOPRSTWX" -> "ȦḂĊḊĖḞĠḢİĿṀṄȮṖṘṠṪẆẊ"
+    val upperDowndot = "BDHILMNORST" -> "ḄḌḤỊḶṂṆỌṚṢṬ"
```
Comment (mathcat4) [inline shared/src/main/scala/SugarMap.scala:8]:
```suggestion
    val lowerUpdot = "abcdefghlmnoprstx" -> "ȧḃċḋėḟġḣŀṁṅȯṗṙṡṫẋ"
    val upperUpdot = "ABCDEFGHILMNOPRSTWX" -> "ȦḂĊḊĖḞĠḢİĿṀṄȮṖṘṠṪẆẊ"
    val upperDowndot = "BDHILMNORST" -> "ḄḌḤỊḶṂṆỌṚṢṬ"
```
well this doesn't seem right

**Review by mathcat4 — CHANGES_REQUESTED** [review]

**Review by mathcat4 — APPROVED** [review]

**Pull request merged**